git-gui/git-gui.sh

```diff
@@ -2203,6 +2203,8 @@ proc do_gitk {revs {is_submodule false}} {
 
 		if {$old_GIT_DIR ne {}} {
 			set env(GIT_DIR) $old_GIT_DIR
+		} else {
+			unset env(GIT_DIR)
```

Comment:
Okay, this handles the non-submodule case when launching a `gitk`, and it affects all programs spawned *after* that `gitk`. May I assume that this was the step your reproduction recipe was missing, and why my Git Bash worked correctly?